src/common/src/tdataformat.c

```diff
@@ -239,6 +251,8 @@ void dataColAppendVal(SDataCol *pCol, const void *value, int numOfRows, int maxP
     if (numOfRows > 0) {
       // Find the first not null value, fill all previouse values as NULL
       dataColSetNEleNull(pCol, numOfRows, maxPoints);
+    } else {
+      tdAllocMemForCol(pCol, maxPoints);
```

Comment:
This problem is known and commented by a TODO note. Because this PR split one big memory allocation process into lots of ones, a lot of function becomes unsafe. A total fixation needs changing 10+ function and I'll do it later.